flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitorTest.java

```diff
@@ -0,0 +1,148 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.contrib.streaming.state;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.MetricOptions;
+import org.apache.flink.metrics.Metric;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.groups.AbstractMetricGroup;
+import org.apache.flink.runtime.metrics.groups.OperatorMetricGroup;
+import org.apache.flink.runtime.metrics.groups.UnregisteredMetricGroups;
+import org.apache.flink.runtime.metrics.scope.ScopeFormats;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.rocksdb.ColumnFamilyHandle;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+
+/**
+ * validate native metric monitor.
+ */
+public class RocksDBNativeMetricMonitorTest {
+
+	private static final String OPERATOR_NAME = "dummy";
+
+	private static final String COLUMN_FAMILY_NAME = "column-family";
+
+	@Test
+	public void testMetricMonitorLifecycle() throws Throwable {
+		//created as a local variable instead of using @Rule
+		//we we can manually control the lifecycle. Using @Rule
+		//but manually calling after causes occasional segmentation
+		//faults.
+		RocksDBResource rocksDBResource = new RocksDBResource();
+		rocksDBResource.before();
+
+		SimpleMetricRegistry registry = new SimpleMetricRegistry();
+		OperatorMetricGroup group = new OperatorMetricGroup(
+			registry,
+			UnregisteredMetricGroups.createUnregisteredTaskMetricGroup(),
+			new OperatorID(),
+			OPERATOR_NAME
+		);
+
+		RocksDBNativeMetricOptions options = new RocksDBNativeMetricOptions();
+		// always returns a non-zero
+		// value since empty memtables
+		// have overhead.
+		options.enableSizeAllMemTables();
+
+		RocksDBNativeMetricMonitor monitor = new RocksDBNativeMetricMonitor(
+			rocksDBResource.getRocksDB(),
+			options,
+			group
+		);
+
+		ColumnFamilyHandle handle = rocksDBResource.createNewColumnFamily(COLUMN_FAMILY_NAME);
+		monitor.registerColumnFamily(COLUMN_FAMILY_NAME, handle);
+
+		Assert.assertEquals("Failed to register metrics for column family", 1, registry.metrics.size());
+
+		RocksDBNativeMetricMonitor.RocksDBNativeMetricView view = registry.metrics.get(0);
+
+		view.update();
+
+		Assert.assertNotEquals("Failed to pull metric from RocksDB", 0, Long.parseLong(view.getValue()));
+
+		view.setValue(0L);
+
+		//removing this line reliably causes segmentation faults
```

Comment:
re-phrase: `After the monitor is closed no metric should be accessing RocksDB anymore. If they do, then this test will likely fail with a segmentation fault.`